### ISSUE TITLE
Pkg REPL compatibility bounds

### ISSUE THREAD

**Issue by staticfloat** [JuliaLang/Pkg.jl#341]
We need a way to specify compatibility bounds when adding packages, interactively through the REPL mode.  I'm thinking something like `pkg> add Compat@[0.55.0+]`, or perhaps `pkg> add Compat` followed by `pkg> restrict Compat 0.55.0-0.66.0`

What do you guys think?

**Comment by KristofferC** [comment]
I mean, you can already do something like

```
(Foo) pkg> add Compat@0.55
  Updating registry at `~/.julia/registries/Uncurated`
  Updating git-repo `https://github.com/JuliaRegistries/Uncurated.git`
 Resolving package versions...
  Updating `Project.toml`
  [34da2185] + Compat v0.55.2
```

Perhaps that should also put `Compat = "0.55"` into the compatibility section?

**Comment by KristofferC** [comment]
Should probably show the `compatibility` bounds in the status output as well

**Comment by staticfloat** [comment]
Maybe we should allow “add Compat@0.55.0-“ to set a lower bound, as that’s the most common use case. Maybe go whole hog and allow “add Compat@0.55.0-0.66.0”

**Comment by StefanKarpinski** [comment]
`Compat = "0.55"` in the compatibility section means `Compat = "^0.55.0"`.

**Comment by oxinabox** [comment]
I propose that the API for this be based around the versions of packages currently being used in the Manifest.
This would be partnered with #1607 which would allow things to ignore the current compat settings.

For sake of example, lets call this function `write_compat`
And it will take options:
 - `extend::Bool` which if true (default?) will *add* the compatibility to what is already there, and if false will clear all current compat sections and just write what you have to be there.
 - `specificy` being one of `^, ~, ==, <=, >=, ` etc (with `^` being the default), which controls what restrictor is written.
- `packages` (optional) which packages to include. Defaults to all packages in Project.toml

So if I just created a new project for some data science I might do:
```
pkg"add CSV, DataFrames, Plots"
write_compat()
```
which will give me a Project.toml withat has those packages and normal (carat) bound to the current versions. Which is what i normally do when I start a new project.

If I want to later restrict to also support a newly released CSV v0.6 I would do:
```
pkg"add CSV@0.6 --ignore-compat"
write_compat("CSV")

pkg"resolve"  # make sure that we no longer are needing to `--ignore-compat`
```

If I became paranoid about thing breaking, and i wanted to lock all the versions down to what i know currently works.
```
write_compat(extend=false, restriction= ==)
```

This API wouldn't do everything, in particular some things would be much easier going into Project.toml to do, like removing compat for an old version of something.

But I think it would nicely handle a lot of common use-cases, and while nontraditional, would be more useful than something that basically just replaces edits to a text file with commands that use basically the same characters in the REPL.

**Comment by IanButterworth** [comment]
Introduced in https://github.com/JuliaLang/Pkg.jl/pull/2702

If anyone's been using it, it would be good to get feedback!
